lib/akamod/select-id.py

```diff
@@ -33,15 +35,37 @@ def selid(body, ctype, prop='handle', use_source='yes'):
     source_name = request_headers.get('Source')
 
     record_id = None
+
     if exists(data, prop):
         v = getprop(data, prop)
         if isinstance(v, basestring):
             record_id = v
         else:
             if v:
+                # Make an array of IDs (if needed) and iterate over it
                 for h in (v if isinstance(v, list) else [v]):
-                    if is_absolute(h):
+                    # If attributes are provided to help select the
+                    # appropriate provider ID value then take the first match
+                    if not record_id and isinstance(h, dict) and attr and \
+                            attrValue:
+                        if h[attr].lower() == attrValue:
+                            record_id = textnode.textnode(h)
+                    # If no attributes are given then select the first
+                    # absolute URI
+                    elif not record_id and is_absolute(h):
                         record_id = h
+                # If no absolute URI was found then take the first value in the
+                # array of original record IDs
+
+                # TODO: This is a gap in the logic which might lead to IDs
+                # TODO: minted with the wrong base ID.
+                #
+                # For example:
```

Comment:
For follow-up with @guegueng we should chat about what the default behavior should be that addresses MO's concerns w/o breaking all other providers.